File: app/java/app/src/main/java/com/example/accounting/domain/model/audit/AuditAction.java
package com.example.accounting.domain.model.audit;

/**
 * 監査操作種別
 */
public enum AuditAction {
    CREATE("作成"),
    UPDATE("更新"),
    DELETE("削除");

    private final String displayName;

    AuditAction(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: app/scala/src/main/scala/com/example/accounting/domain/audit/AuditAction.scala
package com.example.accounting.domain.audit

/** 監査アクション種別 */
enum AuditAction(val code: String, val displayName: String):
  case Create extends AuditAction("CREATE", "作成")
  case Update extends AuditAction("UPDATE", "更新")
  case Delete extends AuditAction("DELETE", "削除")

object AuditAction:
  def fromCode(code: String): AuditAction =
    code match
      case "CREATE" => Create
      case "UPDATE" => Update
      case "DELETE" => Delete
      case _        => throw new IllegalArgumentException(s"Unknown audit action: $code")
